src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -315,7 +303,11 @@ private void expectField(JsonParser jsonParser, String expectedFieldName) throws
     private void expectToken(JsonParser jsonParser, JsonToken expectedToken) throws IOException {
         final JsonToken token = jsonParser.nextToken();
         if (token == null) {
-            throw new EOFException(Thread.currentThread().isInterrupted() ? "Thread was interrupted" : "Stream was closed");
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
Ah ok, thanks for the explanation. I though the `JsonInput` would block and hence `currentThread().interrupt()` would only set the interrupted flag without killing any connection.